src/gun.erl

```diff
@@ -473,7 +475,7 @@ dbg_send_raw(ServerPid, Data) ->
 
 start_link(Owner, Host, Port, Opts) ->
 	proc_lib:start_link(?MODULE, init,
-		[self(), Owner, Host, Port, Opts]).
+        [self(), Owner, Host, Port, Opts], maps:get(conn_timeout, Opts, 1000)).
```

Comment:
Don't really understand this part. We init_ack right at the start so this doesn't actually do anything. The start_link returns before we even try to connect.